rust/arrow/src/array.rs

```diff
@@ -995,6 +995,7 @@ impl From<Vec<(Field, ArrayRef)>> for StructArray {
 
         let data = ArrayData::builder(DataType::Struct(field_types))
             .child_data(field_values.into_iter().map(|a| a.data()).collect())
+            .len(length)
```

Comment:
@sunchao are you happy that we open a JIRA to calculate null struct values in future as the `bitor` of the array's bitmasks?

Reply:
I think we need to decide whether an incoming vector such as:
```
[
  (BooleanType, [false, null, true, null, false]),
  (Int32Type, [32, null, null, null, 28])
]
```
Should be converted to:
```
(StructType, [(false, 32), null, (true, null), null, (false, 28)])
```

or:
```
(StructType, [(false, 32), (null, null), (true, null), (null, null), (false, 28)])
```

These two may have subtle differences. Feel free to open a JIRA for this and maybe we can have more discussion there.

Reply:
Thanks, I'll open a JIRA. I'm curious to know how C++ does things, I'll maybe also ask in the mailing list post 0.13 release

Reply:
I think this is unique in Rust - C++ doesn't have a API to convert a vector of `(Field, ArrayRef)` into a `StructArray`.